File: consumer/src/test/java/ru/romanow/protocols/consumer/service/OperationServiceTest.java
package ru.romanow.protocols.consumer.service;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.cloud.contract.stubrunner.spring.AutoConfigureStubRunner;
import org.springframework.cloud.contract.stubrunner.spring.StubRunnerProperties;
import org.springframework.test.context.ActiveProfiles;
import org.springframework.test.context.junit4.SpringRunner;
import ru.romanow.protocols.api.model.TestObjectRequest;
import ru.romanow.protocols.api.model.TestObjectResponse;
import ru.romanow.protocols.consumer.exception.RestRequestException;

import static org.junit.Assert.assertEquals;

@ActiveProfiles("test")
@SpringBootTest
@RunWith(SpringRunner.class)
@AutoConfigureStubRunner(
        ids = "ru.romanow.protocols:producer:+:stubs:8080",
        stubsMode = StubRunnerProperties.StubsMode.LOCAL)
public class OperationServiceTest {
    private static final Integer CODE = 100;
    private static final String DATA = "TEST";

    @Autowired
    private OperationService operationService;

    @Test
    public void testMakeOperation() {
        final int id = 101;
        final String searchString = "test";
        TestObjectRequest request = new TestObjectRequest(id, searchString);
        TestObjectResponse response = operationService.makeOperation(request);

        assertEquals(response.getCode(), CODE);
        assertEquals(response.getData(), DATA);
    }

    @Test(expected = RestRequestException.class)
    public void testMakeOperationFail() {
        final int id = 1;
        final String searchString = "test";
        TestObjectRequest request = new TestObjectRequest(id, searchString);
        TestObjectResponse response = operationService.makeOperation(request);
    }
}

File: consumer/src/test/java/ru/romanow/protocols/consumer/service/OperationServiceTest.kt
package ru.romanow.protocols.consumer.service

import org.assertj.core.api.Assertions.assertThat
import org.assertj.core.api.AssertionsForClassTypes.assertThatThrownBy
import org.junit.jupiter.api.Test
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.boot.test.context.SpringBootTest
import org.springframework.cloud.contract.stubrunner.spring.AutoConfigureStubRunner
import org.springframework.cloud.contract.stubrunner.spring.StubRunnerProperties
import org.springframework.test.context.ActiveProfiles
import ru.romanow.protocols.api.model.TestObjectRequest
import ru.romanow.protocols.consumer.exception.RestRequestException

@ActiveProfiles("test")
@SpringBootTest
@AutoConfigureStubRunner(
    ids = ["ru.romanow.protocols:producer:+:stubs:8080"],
    stubsMode = StubRunnerProperties.StubsMode.LOCAL
)
internal class OperationServiceTest {

    @Autowired
    private lateinit var operationService: OperationService

    @Test
    fun testMakeOperation() {
        val request = TestObjectRequest(ID, SEARCH_STRING)
        val response = operationService.makeOperation(request)
        assertThat(response.code).isEqualTo(CODE)
        assertThat(response.data).isEqualTo(DATA)
    }

    @Test
    fun testMakeOperationFail() {
        val request = TestObjectRequest(ID, SEARCH_STRING)
        assertThatThrownBy {
            operationService.makeOperation(request)
        }.isInstanceOf(RestRequestException::class.java)
    }

    companion object {
        private const val ID = 1
        private const val SEARCH_STRING = "test"
        private const val CODE = 100
        private const val DATA = "TEST"
    }
}